baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/manager/DecisionLifecycleOrchestrator.java

```diff
@@ -129,9 +117,11 @@ public DecisionVersion createOrUpdateVersion(String customerId, DecisionRequest
         DecisionVersion decisionVersion = decisionManager.createOrUpdateVersion(customerId, decisionRequest);
 
         if (decisionVersion.getKafkaConfig() != null) {
-            Credential credential = getCustomerCredential(customerId, decisionRequest);
+            KafkaService kafkaClient = kafkaClientFactory.createKafkaClient();
+            Credential credential = kafkaClient.getCustomerCredential(customerId);
```

Comment:
The only problem I see here is that we have already persisted the `DecisionRequest` by this point, so we would end up with a bunch of Decisions in the users account that we can't support and they have to manually delete. I think ideally if the user requests a Decision with Kafka Topics attached and we are in an environment that doesn't support Kafka, we wouldn't even persist the request, we'd just return the 400 Bad Request.

Couple of potential solutions I can see here:

- Provide a custom validator (perhaps integrated into Hibernate validator) that calls the `KafkaService` to verify that the environment can support Kafka integration if the user has requested it.
- Do a check here before any call to persist the `DecisionRequest` that checks to see if that if the user wants Kafka attached, we can support it in this environment.

Reply:
In existing FM, kafka is by default supported, but the Kafka credential collection for a decision could still fail after the decisionreques has persisted. How do we solve this seemly dangling case?

Reply:
@myeung18 That's a good point. The answer is that we don't.

I do think the scenarios are slightly different:

1. If Kafka is enabled and the user requests Kafka attached to their Decision but we fail to provision the Service Account for Kafka, this should be recorded as a failure to deploy with a meaningful error code returned to the user. We should also have monitoring alerts configured to watch for errors in creating the Service Account for Kafka occurring in the logs so that we can investigate and fix any potential problem.
2. If Kafka is disabled, it's a validation error if the user requests a Decision with Kafka attached and we should fail-fast without persisting anything.

So I think for scenario 1, we need to ensure that we properly log and capture the failure to provision a Service Account for Kafka. I will create a JIRA issue for that.

For this user story, I do think we should not persist anything if the user is requesting Kafka but it is disabled for that environment. WDYT?

Reply:
@robpblake 
I think what you mentioned made sense.  That should be the way to be implemented.